tests/integration/test_tls.py

```diff
@@ -158,6 +159,9 @@ async def test_rotate_tls_key(ops_test: OpsTest) -> None:
     # set key using auto-generated key for each unit
     for unit in ops_test.model.applications[app].units:
         action = await unit.run_action(action_name="set-tls-private-key")
+        # action.wait not always await action status to change
+        # adding some time to address finicky CI test
+        sleep(4)
```

Comment:
is it possible to use retries instead of sleep here? im always wary of sleeping in tests

Reply:
sure, I think I prefer that also

Reply:
8e9d84b